edgelet/iotedged/src/windows.rs

```diff
@@ -46,26 +45,18 @@ fn iotedge_service_main(args: Vec<OsString>) {
 }
 
 fn run_as_service(_: Vec<OsString>) -> Result<ServiceStatusHandle, Error> {
-    // setup a channel for notifying service stop/shutdown
-    let (sender, receiver) = oneshot::channel();
-    let sender = RefCell::new(Some(sender)); // register() takes Fn, not FnMut
-
     // setup the service control handler
     let status_handle = register(
         IOTEDGED_SERVICE_NAME,
         move |control_event| match control_event {
             ServiceControl::Shutdown | ServiceControl::Stop => {
                 info!("{} service is shutting down", IOTEDGED_SERVICE_NAME);
 
-                // If sender is None, then it has already been consumed by a previous shutdown / stop notification
-                // that signaled the receiver. There's nothing more to do in that case.
-                if let Some(sender) = sender.borrow_mut().take() {
-                    sender.send(()).unwrap_or_else(|err| {
-                        error!(
-                            "An error occurred while raising service shutdown signal: {:?}",
-                            err
-                        );
-                    });
+                // We were told by windows to shutdown :(
+                // Let's generate a CTRL_C event to trigger the shutdown sequence
+                let process_id = process::id();
+                unsafe {
+                    GenerateConsoleCtrlEvent(CTRL_C_EVENT, process_id as DWORD);
```

Comment:
Looking at the doc for [GenerateConsoleCtrlEvent](https://docs.microsoft.com/en-us/windows/console/generateconsolectrlevent) it looks like we may be able to get away with passing zero for the process ID.

Reply:
I'll test this on my Windows box